src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -380,14 +380,14 @@ public static string FormatDouble(double value, string? format, NumberFormatInfo
         {
             Span<byte> pDigits = stackalloc byte[DoubleNumberBufferLength];
             var sb = new ValueStringBuilder(stackalloc char[CharStackBufferSize]);
-            return FormatDouble(ref sb, pDigits, value, format, info) ?? sb.ToString();
+            return FormatDouble(ref sb, ref pDigits, value, format, info) ?? sb.ToString();
```

Comment:
I followed the same pattern used for ValueStringBuilder (passing by ref), but I'm not sure if it improve the performance in case of Span. I can't find other APIs using ref for Spans. The ASM code is bigger when using ref: [SharpLab Ref Span](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHufv6nySAQAYGAMWj5sGACIQArsAA2MABQATBcpgMAbtiXyYASi68enauesMAygAdsAOwA8wAJ4YYAPgb2ZAJYA5gEYuAwAvAy4GNhgANYGShBgDB5eANrkwsIAugDcZjY8xADs4pLScooqqgFOfoEhYcaFVjYAvkUM3cX2UAH6XgJCxNkVUFKyWrX1do6u6T6NwaG4pu3WlsXFZQxO8kpKbcVdmz3nvSWMY6PjEpNVMzCkGs96BkYbxds73A7ONyeZb+VZhSLRWIJJIpNLArI5ApXax7B5TaraV4rZrrE6dbrI8z9QbSHS3EQTdHPV4AxbA3ygnHfGy/P68PYHI546xnDpAA===)

Reply:
ValueStringBuilder  is passed by ref for functional correctness; not for performance. Span should not be passed by ref.

Reply:
Reverted